src/systemd/sd-network.h

```diff
@@ -94,6 +94,15 @@ int sd_network_link_get_setup_state(int ifindex, char **state);
  */
 int sd_network_link_get_operational_state(int ifindex, char **state);
 
+/* Indicates whether the network is relevant to being online.
+ * Possible states:
+ *   true: the connection is required to consider the system online
+ *   no: the connection is not required
+ * Possible return codes:
+ *   -ENODATA: networkd is not aware of the link
+ */
+int sd_network_link_get_required_online(int ifindex, char **required);
```

Comment:
required_*for*_online

I don't think this needs to return a char**. Just make it return 1 if yes, 0 if no, <0 on error.